code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -322,18 +325,19 @@
 	if(parents[2])
 		nullify_pipenet(parents[2])
 
-/obj/machinery/atmospherics/components/binary/thermomachine/attackby_secondary(obj/item/item, mob/user, params)
-	. = ..()
-	if(panel_open && item.tool_behaviour == TOOL_WRENCH && !check_pipe_on_turf())
-		if(default_unfasten_wrench(user, item))
-			return SECONDARY_ATTACK_CONTINUE_CHAIN
-	if(panel_open && item.tool_behaviour == TOOL_MULTITOOL)
+/obj/machinery/atmospherics/components/binary/thermomachine/wrench_act_secondary(mob/living/user, obj/item/tool)
+	if(panel_open && !check_pipe_on_turf() && default_unfasten_wrench(user, tool))
+		return TRUE
+	return FALSE
```

Comment:
I have a feeling this would be easier to reason with like this
```suggestion
	if(!panel_open || check_pipe_on_turf())
		return FALSE
	if(default_unfasten_wrench(user, tool))
		return TRUE
	return FALSE
```